src/blake2b.c

```diff
@@ -1,19 +1,22 @@
 #include "blake2b.h"
 
+#include <ledger_assert.h>
 #include <stdint.h>
 #include <string.h>
 
+#include "sia.h"
+
 void blake2b_init(cx_blake2b_t *S) {
-    cx_blake2b_init(S, 256);
+    LEDGER_ASSERT(CX_OK == cx_blake2b_init_no_throw(S, 256), "blake2b_init failed");
 }
 
 void blake2b_update(cx_blake2b_t *S, const uint8_t *in, uint64_t inlen) {
-    cx_hash((cx_hash_t *)S, 0, in, inlen, NULL, 0);
+    LEDGER_ASSERT(CX_OK == cx_hash_no_throw((cx_hash_t *)S, 0, in, inlen, NULL, 0), "blake2b_update failed");
 }
 
 void blake2b_final(cx_blake2b_t *S, uint8_t *out, uint64_t outlen) {
-    uint8_t buf[32];
-    cx_hash((cx_hash_t *)S, CX_LAST, NULL, 0, buf, sizeof(buf));
+    uint8_t buf[32] = {0};
```

Comment:
Why using buf as an intermediate buffer? In the memmove below you still need to check the outlen value, would be simpler without this buffer.

Reply:
Because it appears cx_hash_no_throw returns an error if outlen < 32.

Reply:
I'd say you should add a check, maybe an assert, that indeed outlen <= 32 for the memcpy below.